spec/std/http/headers_spec.cr

```diff
@@ -206,4 +206,28 @@ describe HTTP::Headers do
     headers.add?("foobar", invalid_value).should be_false
     headers.add?("foobar", [invalid_value]).should be_false
   end
+
+  describe "after serialization" do
+    it "raises on modification (#8712)" do
+      headers = HTTP::Headers.new
+      io = IO::Memory.new
+      headers.serialize(io)
+
+      expect_raises(::Exception) { headers["Foo"] = "Bar" }
+      expect_raises(::Exception) { headers["Foo"] = ["Bar", "Baz"] }
+      expect_raises(::Exception) { headers.add "Foo", "Bar" }
+      expect_raises(::Exception) { headers.add "Foo", ["Bar", "Baz"] }
+      expect_raises(::Exception) { headers.add? "Foo", "Bar" }
+      expect_raises(::Exception) { headers.add? "Foo", ["Bar", "Baz"] }
+      expect_raises(::Exception) { headers.delete "Foo" }
+      expect_raises(::Exception) { headers.merge!({"Foo" => "Bar"}) }
```

Comment:
The expectation should be more specific. With `expect_raises(Exception)` the code could be completely broken and raise an unrelated expection, yet still the spec passes.
I don't think there's a specific error class for this, and it's probably not worth adding a dedicated one. But maybe `IO::Error` would be fine. And please check the error message.

Reply:
I changed it to `IO::Error` and checked also the error message